src/acl_auto_configure.cpp

```diff
@@ -193,6 +193,125 @@ static int read_string_counters(const std::string &str,
   return result != "";
 }
 
+static bool read_kernel_arg_info(const std::string &config_str,
+                                 bool kernel_arg_info_available,
+                                 std::string::size_type &curr_pos,
+                                 acl_kernel_arg_info_t &arg_info,
+                                 std::vector<int> &counters) noexcept {
+  bool result = true;
+  auto addr_space_type = 0U;
+  auto category = 0U;
+  auto size = 0U;
+  auto num_buffer_locations = 0U;
+  int total_fields_arguments = 0;
+  if (result) {
+    result = result && read_int_counters(config_str, curr_pos,
+                                         total_fields_arguments, counters);
+  }
+  counters.emplace_back(total_fields_arguments);
+  unsigned alignment = ACL_MEM_ALIGN; // Set default to 1024 bytes
+  result =
+      result &&
+      read_uint_counters(config_str, curr_pos, addr_space_type, counters) &&
+      read_uint_counters(config_str, curr_pos, category, counters) &&
+      read_uint_counters(config_str, curr_pos, size, counters);
+  if (result) {
+    result =
+        result && read_uint_counters(config_str, curr_pos, alignment, counters);
+  }
+
+  std::string buffer_location = "";
+  if (result) {
+    result = result && read_uint_counters(config_str, curr_pos,
+                                          num_buffer_locations, counters);
+    for (auto k = 0U; result && (k < num_buffer_locations); k++) {
+      if (k /* FIXME j */ > 0) {
+        buffer_location = buffer_location + " ";
+      }
+
+      result = result && read_string_counters(config_str, curr_pos,
+                                              buffer_location, counters);
```

Comment:
This actually overwrites `buffer_location`, instead of appending as likely intended.

Reply:
This has been resolved in #108